.github/workflows/main.yml

```diff
@@ -0,0 +1,347 @@
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: NetBeans
+
+on:
+  push:
+  pull_request:
+
+jobs:
+    
+# primary build job, most other jobs use the artifact produced here
+# artifact is only produced once in the matrix
+  base-build:
+    name: Build Clusters on JDK ${{ matrix.java }} 
+    runs-on: ubuntu-latest
+    env:
+      ANT_OPTS: -Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json
+    strategy:
+      matrix:
+        java: [ '11', '17' ]
+      fail-fast: false
+    steps:
+        
+      - name: Set up JDK ${{ matrix.java }} 
+        uses: actions/setup-java@v2
+        with:
+          java-version: ${{ matrix.java }} 
+          distribution: 'zulu'
+
+      - name: Caching dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.hgexternalcache
+          key: ${{ runner.os }}-${{ hashFiles('*/external/binaries-list', '*/*/external/binaries-list') }}
+          restore-keys: ${{ runner.os }}-
+          
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Build NetBeans
+        run: ant -Dcluster.config=release build-nozip
+
+      - name: Archive Build
+        if: ${{ matrix.java == '11' }}
+        run: tar -czf /tmp/build.tar.gz --exclude ".git" .
+
+      - name: Upload Build
+        if: ${{ matrix.java == '11' }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: build
+          path: /tmp/build.tar.gz
+          retention-days: 1
+          if-no-files-found: error
+
+
+# secondary jobs
+  linux-commit-validation:
+    needs: base-build
+    name: Commit Validation on Linux/JDK ${{ matrix.java }} 
+    runs-on: ubuntu-latest
+    env:
+      ANT_OPTS: -Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json
+    strategy:
+      matrix:
+        java: [ '8', '11', '17', '19-ea' ]
+      fail-fast: false
+    steps:
+
+      - name: Set up JDK ${{ matrix.java }} 
+        uses: actions/setup-java@v2
+        with:
+          java-version: ${{ matrix.java }} 
+          distribution: 'zulu'
+
+      - name: Download Build
+        uses: actions/download-artifact@v3
+        with:
+          name: build
+
+      - name: Extract
+        run: tar -xzf build.tar.gz --exclude nbbuild/build/test
+        
+      - name: JMS config
+        if: ${{ matrix.java != '8' }}
+        run: echo "JAVA_TOOL_OPTIONS=--add-opens=java.base/java.net=ALL-UNNAMED --add-opens=java.desktop/javax.swing=ALL-UNNAMED" >> $GITHUB_ENV
```

Comment:
19-ea fails currently. It appears JDK 19 requires now `-Djava.security.manager=allow` to be set, since the warning became now a runtime exception:
```
[junit] java.lang.UnsupportedOperationException: The Security Manager is deprecated and will be removed in a future release
    [junit] 	at java.base/java.lang.System.setSecurityManager(System.java:416)
    [junit] 	at org.netbeans.TopSecurityManager.install(TopSecurityManager.java:525)
    [junit] 	at org.netbeans.core.NbLifecycleManager.advancePolicy(NbLifecycleManager.java:69)
    [junit] 	at org.netbeans.core.GuiRunLevel.run(GuiRunLevel.java:84)
    [junit] 	at org.netbeans.core.startup.Main.start(Main.java:312)
    [junit] 	at org.netbeans.core.startup.TopThreadGroup.run(TopThreadGroup.java:98)
    [junit] 	at java.base/java.lang.Thread.run(Thread.java:828)
```
but even if the flag is set (tested it locally) it would fail later:
```
 [nb-javac] Compiling 1 source file to /home/mbien/NetBeansProjects/netbeans/java/java.hints/build/classes
 [nb-javac] /home/mbien/NetBeansProjects/netbeans/java/java.hints/src/org/netbeans/modules/java/hints/bugs/Unused.java:82: error: cannot find symbol
 [nb-javac]                 fix = JavaFixUtilities.safelyRemoveFromParent(ctx, Bundle.FIX_RemoveUsedElement(name), ud.unusedElementPath);
 [nb-javac]                                       ^
 [nb-javac]   symbol:   method safelyRemoveFromParent(HintContext,String,TreePath)
 [nb-javac]   location: class JavaFixUtilities
 [nb-javac] 1 error

```
I will swap 19-ea with 18 for now since more work is needed there.

Reply:
19-ea job is now green too

Reply:
found the OpenJDK PR: https://github.com/openjdk/jdk/pull/5204